OpenRA.Mods.Common/Activities/FindResources.cs

```diff
@@ -103,7 +103,7 @@ public override Activity Tick(Actor self)
 				foreach (var n in notify)
 					n.MovingToResources(self, closestHarvestablePosition.Value, this);
 
-				return ActivityUtils.SequenceActivities(mobile.MoveTo(closestHarvestablePosition.Value, 1), new HarvestResource(self), this);
+				return ActivityUtils.SequenceActivities(mobile.MoveTo(closestHarvestablePosition.Value, 2), new HarvestResource(self), this);
```

Comment:
What I've done is, I added the diagonal cells (and more horizontal and diagonal cells).
It CAN move to an empty cell even when the parameter == 0, because of the way claiming works: they don't respect enemy claim. There's already a check to see if the cell is mineable by the time the harvester reaches there and searches for new resources when the cell is empty.